lisa/tests/staging/scheduler/util_tracking.py

```diff
@@ -0,0 +1,338 @@
+                cls.run_rtapp(target, res_dir, rtapp_profile, ftrace_coll)
+
+        return cls(res_dir, plat_info)
+
+class Convergence(UtilTrackingBase):
+    """
+    Basic checks for estimated utilization signals
+
+    **Expected Behaviour:**
+
+    The estimated utilization of a task is properly computed starting form its
+    `util_avg` value at the end of each activation.
+
+    Two signals composes the estimated utlization of a task:
+
+    * `util_est_enqueued` : is expected to match the max between `util_avg` and
+      `util_est_ewma` at the end of the previous activation
+
+    * `util_est_ewma` : is expected to track an Exponential Weighted Moving
+      Average of the `util_avg` signal sampled at the end of each activation.
+
+    Based on these two invariant, this class provides a set of tests to verify
+    these conditions using different methods and sampling points.
+    """
+
+    PhaseStats = namedtuple("PhaseStats", [
+        'start', 'end', 'area_util', 'area_enqueued', 'area_ewma'])
+
+    ActivationSignals = namedtuple("ActivationSignals", [
+        'time', 'util_avg', 'util_est_enqueued', 'util_est_ewma'])
+
+    @classmethod
+    def get_rtapp_profile(cls, plat_info):
+        big_cpu = plat_info["capacity-classes"][-1][0]
+
+        task = (
+            # Big task: 15 activations at full utilization
+            Periodic(
+                duty_cycle_pct=75,
+                duration_s=5,
+                period_ms=200,
+                cpus=[big_cpu]) +
+            # Ramp Down
+            Ramp(
+                start_pct=50,
+                end_pct=5,
+                delta_pct=20,
+                time_s=1,
+                period_ms=200,
+                cpus=[big_cpu]) +
+            # Ramp Up
+            Ramp(
+                start_pct=10,
+                end_pct=60,
+                delta_pct=20,
+                time_s=1,
+                period_ms=200,
+                cpus=[big_cpu])
+        )
+        return {'test_task' : task}
+
+    @property
+    @memoized
+    def fast_ramp(self):
+        # The EWMA "fast ramp up" feature has been merged only in kernel TBD
+        # TODO fix kernel version
+        return False
+
+        min_kernel = KernelVersion('4.5-rc1').parts
+        cur_kernel = self.plat_info['kernel']['version'].parts
+        return cur_kernel < min_kernel
+
+    def _plot_signals(self, task, test, failures):
+        signals = ['util', 'util_est_enqueued', 'util_est_ewma']
+        ax = self.trace.analysis.load_tracking.plot_task_signals(task, signals=signals)
+        ax = self.trace.analysis.rta.plot_phases(task, axis=ax);
+        for start in failures:
+            ax.axvline(start, alpha=0.5, color='r', axis=ax)
+        filepath = os.path.join(self.res_dir, 'util_est_{}.png'.format(test))
+        self.trace.analysis.rta.save_plot(ax.figure, filepath=filepath)
+
+    @requires_events('sched_util_est_task', 'sched_load_se')
```

Comment:
I was going to say that should be `df_tasks_signal.used_events` but it's a bit trickier than this... Ohwell.